jbmc/src/java_bytecode/java_utils.cpp

```diff
@@ -446,184 +445,222 @@ const std::unordered_set<std::string> cprover_methods_to_ignore
+  // we have the base class, java.lang.Object, length and data
+  // of appropriate type
+  class_type.set_tag(symbol_type.get_identifier());
+  // Note that non-array types don't have "java::" at the beginning of their
+  // tag, and their name is "java::" + their tag. Since arrays do have
+  // "java::" at the beginning of their tag we set the name to be the same as
+  // the tag.
+  class_type.set_name(symbol_type.get_identifier());
+
+  class_type.components().reserve(3);
+  class_typet::componentt base_class_component(
+    "@java.lang.Object", symbol_typet("java::java.lang.Object"));
+  base_class_component.set_pretty_name("@java.lang.Object");
+  base_class_component.set_base_name("@java.lang.Object");
+  class_type.components().push_back(base_class_component);
+
+  class_typet::componentt length_component("length", java_int_type());
+  length_component.set_pretty_name("length");
+  length_component.set_base_name("length");
+  class_type.components().push_back(length_component);
+
+  class_typet::componentt data_component(
+    "data", java_reference_type(java_type_from_char(l)));
+  data_component.set_pretty_name("data");
+  data_component.set_base_name("data");
+  class_type.components().push_back(data_component);
+
+  class_type.add_base(symbol_typet("java::java.lang.Object"));
+
+  INVARIANT(
+    is_valid_java_array(class_type),
+    "Constructed a new type representing a Java Array "
+    "object that doesn't match expectations");
+
+  symbolt symbol;
+  symbol.name = symbol_type_identifier;
+  symbol.base_name = symbol_type.get(ID_C_base_name);
+  symbol.is_type = true;
+  symbol.type = class_type;
+  symbol.mode = ID_java;
+  symbol_table.add(symbol);
+
+  // Also provide a clone method:
+  // ----------------------------
+
+  const irep_idt clone_name = identifier + +".clone:()Ljava/lang/Object;";
+  code_typet::parametert this_param;
+  this_param.set_identifier(id2string(clone_name) + "::this");
+  this_param.set_base_name("this");
+  this_param.set_this();
+  this_param.type() = java_reference_type(symbol_type);
+  const code_typet clone_type({this_param}, java_lang_object_type());
+
+  parameter_symbolt this_symbol;
+  this_symbol.name = this_param.get_identifier();
+  this_symbol.base_name = this_param.get_base_name();
+  this_symbol.pretty_name = this_symbol.base_name;
+  this_symbol.type = this_param.type();
+  this_symbol.mode = ID_java;
+  symbol_table.add(this_symbol);
+
+  const irep_idt local_name = id2string(clone_name) + "::cloned_array";
+  auxiliary_symbolt local_symbol;
+  local_symbol.name = local_name;
+  local_symbol.base_name = "cloned_array";
+  local_symbol.pretty_name = local_symbol.base_name;
+  local_symbol.type = java_reference_type(symbol_type);
+  local_symbol.mode = ID_java;
+  symbol_table.add(local_symbol);
+  const auto &local_symexpr = local_symbol.symbol_expr();
+
+  code_blockt clone_body;
+
+  code_declt declare_cloned(local_symexpr);
+  clone_body.move_to_operands(declare_cloned);
```

Comment:
code_blockt has a move method

Reply:
will change, but this is old code that just moved